File: src/main/java/kr/co/bookand/backend/account/domain/dto/RevokeDto.java
package kr.co.bookand.backend.account.domain.dto;


import lombok.Builder;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

public class RevokeDto {


    public record RevokeReasonRequest(
            @NotNull
            @NotEmpty
            String revokeType,

            String reason,
            @NotNull
            @NotEmpty
            String socialAccessToken
    ) {
        @Builder
        public RevokeReasonRequest {
        }
    }
}

File: src/main/kotlin/kr/co/bookand/backend/account/domain/dto/RevokeDto.kt
package kr.co.bookand.backend.account.domain.dto

import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class RevokeReasonRequest(
    @NotNull
    @NotEmpty
    val revokeType: String,

    val reason: String,

    @NotNull
    @NotEmpty
    val socialAccessToken: String
)